### ISSUE TITLE
Metrics cannot be enabled from a config file

### ISSUE THREAD

**Issue by chfast** [ethereum/go-ethereum#24178]
#### System information

Geth version: 1.10.14
OS & Version: Linux

#### Expected behaviour

Metrics (as with `--metrics`) should be enabled by the entry in the config file `[Metrics] Enabled = true`.

#### Actual behaviour

Metrics are not enabled.

#### Steps to reproduce the behaviour

1. Dump config for metrics.
   ```
   geth --metrics dumpconfig
   ```
   You will geth section as
   ```
   [Metrics]
   Enabled = true
   HTTP = "127.0.0.1"
   Port = 6060
   InfluxDBEndpoint = "http://localhost:8086"
   InfluxDBDatabase = "geth"
   InfluxDBUsername = "test"
   InfluxDBPassword = "test"
   InfluxDBTags = "host=localhost"
   ```
2. Try if metrics work at all.
   ```
   geth --metrics
   INFO [01-02|11:28:29.998] Starting Geth on Ethereum mainnet... 
   INFO [01-02|11:28:29.998] Enabling metrics collection 
   ```
3. Try with config file.
   ```
   geth --config metrics.toml
   ```
   The log "Enabling metrics collection" does not show up. The server at localhost:6060 does not work.

**Comment by s1na** [comment]
There is an old PR for this: https://github.com/ethereum/go-ethereum/pull/23029. I'll polish it up, make it merge-able.

**Comment by s1na** [comment]
This is not as straightforward as I thought. Problem is the metrics need to be enabled super early on start: https://github.com/ethereum/go-ethereum/blob/335914a63afae019a2b901d88f13b28eb0d6ca58/metrics/metrics.go#L45

Config file processing happens much later on. Only way I see is to manually extract the enable flag from the config file in `metrics.init()` which adds on top of the hack.

I'll put this issue on hold until there's a better idea how to handle it.

**Comment by zzh1996** [comment]
@MariusVanDerWijden I run into this issue today. It isn't a single flag. For me it's 7 flags: `--metrics --metrics.influxdb --metrics.influxdb.endpoint --metrics.influxdb.database --metrics.influxdb.username --metrics.influxdb.password --metrics.influxdb.tags`.

Please reopen this issue.

**Comment by zzh1996** [comment]
@holiman No, you are wrong. I already tried that and it doesn't work.
I also checked the source code. Only the cmd line will be used.
https://github.com/ethereum/go-ethereum/blob/v1.11.3/cmd/geth/main.go#L326

**Comment by zzh1996** [comment]
@holiman If you decide not to fix this in the next release, please remove these flags from the config file. It's very misleading.

**Comment by 0x416e746f6e** [comment]
I agree, this is misleading.  `geth` could at least have logged something.
